build-tools/packages/build-cli/src/commands/release.ts

```diff
@@ -102,733 +55,28 @@ export class ReleaseCommand<T extends typeof ReleaseCommand.flags>
-
-            case "CheckInstallBuildTools": {
-                const installQuestion: inquirer.ConfirmQuestion = {
-                    type: "confirm",
-                    name: "install",
-                    message: `Do you want to install the Fluid build-tools? You don't need to do this if you installed them globally.`,
-                };
-
-                const answer = await inquirer.prompt(installQuestion);
-                if (answer.install === true) {
-                    this.log(`Installing build-tools so we can run build:genver`);
-                    const buildToolsMonoRepo = context.repo.releaseGroups.get(
-                        MonoRepoKind.BuildTools,
-                    )!;
-                    const ret = await buildToolsMonoRepo.install();
-                    if (ret.error) {
-                        this.errorLog("Install failed.");
-                        this.machine.action("failure");
-                    }
-                } else {
-                    this.warn(`Skipping installation.`);
-                }
-
-                this.machine.action("success");
-                break;
-            }
-
-            case "CheckMainNextIntegrated": {
-                // TODO: Implement this
-                if (this.bumpType === "major") {
-                    if (this.shouldCheckMainNextIntegrated) {
-                        this.warn(
-                            chalk.red(`Automated main/next integration check not yet implemented.`),
-                        );
-                        this.warn(
-                            `Make sure next has been integrated into main before continuing.`,
-                        );
-
-                        const confirmIntegratedQuestion: inquirer.ConfirmQuestion = {
-                            type: "confirm",
-                            name: "integrated",
-                            message: `Has next has been integrated into main?`,
-                        };
-
-                        const answers = await inquirer.prompt(confirmIntegratedQuestion);
-                        if (answers.integrated !== true) {
-                            this.machine.action("failed");
-                        }
-                    } else {
-                        this.warn("Skipping main/next integration check.");
-                    }
-                }
-
-                this.machine.action("success");
-                break;
-            }
-
-            case "CheckReleaseIsDone": {
-                const wasReleased = await isReleased(
-                    context,
-                    this.releaseGroup!,
-                    this.releaseVersion!,
-                );
-                if (wasReleased) {
-                    this.machine.action("success");
-                } else {
-                    this.machine.action("failure");
-                }
-
-                break;
-            }
+        const handler = new FluidReleaseStateHandler(this.machine, this.logger);
```

Comment:
Can we inline this with the `this.handler` assignment?

Reply:
Good catch.